### PR TITLE
Api version header

### LINKED ISSUE #516: Catalog request not sending Broker API version
For our implementation of the ansible-service-broker (https://github.com/fusor/ansible-service-broker/) we require that all requests have a header specifying the broker API version. (see here for an example request using this header: https://github.com/fusor/ansible-service-broker/blob/master/test/catalog.sh#L5).

Without the header the catalog request 404s and fails to bind the broker to the service catalog.

### LINKED ISSUE #569: provisioning, deprovisioning, binding and unbinding do not handle auth
There are 4 comments in `./pkg/brokerapi/openservicebroker/open_service_broker_client.go` that read: `// TODO: Handle the auth`. These need to be replaced by code that:

- Walks from the binding/instance up to the broker
- Fetches the broker's auth credentials
- Uses the auth credentials in the provision/deprovision/bind/unbind request

### PR DESCRIPTION
**Pull request by MHBauer**
Fixes #516
Fixes #569 

also sets the auth header

### CODE REVIEW

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 1]
```diff
@@ -279,7 +282,7 @@ func (c *openServiceBrokerClient) pollBroker(ID string, operation string) error
 	pollingURL := fmt.Sprintf(pollingFormatString, c.url, ID)
 	for i := 0; i < pollingAmountLimit; i++ {
 		glog.V(3).Infof("Polling broker %v at %s attempt %v", c.name, pollingURL, i+1)
-		pollResp, err := util.SendRequest(c.client, http.MethodGet, pollingURL, pollReq)
+		pollResp, err := sendOSBRequest(c, http.MethodGet, pollingURL, pollReq)
```
Comment (arschles) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: do we need to send auth headers for the polling endpoint (`last_operation`)?

Comment (MHBauer) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: from 
https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#authentication
```
The marketplace must authenticate with the service broker using HTTP basic authentication (the Authorization: header) on every request. 
```

Comment (arschles) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: perfect, thanks

**Comment by arschles** [comment]
LGTM

**Comment by arschles** [comment]
This PR supersedes https://github.com/kubernetes-incubator/service-catalog/pull/563

**Comment by arschles** [comment]
I am gonna merge this now, as it fixes a blocking MVP 2 bug.

**Pull request merged**